docs/doc_example/main.cpp

```diff
@@ -6,29 +6,24 @@
 
 #include "src/Runtime/ExecutionSession.hpp"
 
-int main() {
-  // Read compiler options from environment first, and ensure -O3 is used.
-  int rc;
-  std::cout << "Start compiling add.onnx" << std::endl;
-  rc = onnx_mlir::omSetCompilerOptionsFromEnv("TEST_DOC_EXAMPLE");
-  if (rc != onnx_mlir::CompilerSuccess) {
-    std::cerr << "Failed to process TEST_DOC_EXAMPLE compiler options with "
-                 "error code "
-              << rc << "." << std::endl;
-    return rc;
-  }
-  rc = onnx_mlir::omSetCompilerOption(
-      onnx_mlir::OptionKind::CompilerOptLevel, "3");
-  if (rc != onnx_mlir::CompilerSuccess) {
-    std::cerr << "Failed to process -O3 compiler option with error code " << rc
-              << "." << std::endl;
-    return rc;
-  }
-  // Compile the doc example into a model library.
+int main(int argc, char *argv[]) {
+  // Read compiler options from command line and compile the doc example into a
+  // model library.
   const char *errorMessage = NULL;
   const char *compiledFilename;
-  rc = onnx_mlir::omCompileFromFile("add.onnx", "add-cppinterface",
-      onnx_mlir::EmitLib, &compiledFilename, &errorMessage);
+  std::string commandLineStr = "";
+  if (std::string(argv[0]) != "onnx-mlir") {
+    std::cerr << "Wrong program name received." << std::endl;
+    return -1;
+  }
+  for (int i = 1; i < argc; i++) {
```

Comment:
Just to simplify the reading of the code, you could isolate the processing of the argc and args into a function that returns a single std::string. It would help the reading of the code.

you have it already below: `readArgsFromCommandLine`... just copy and paste